src/linter/ui5Types/fixHints/JquerySapFixHintsGenerator.ts

```diff
@@ -774,13 +774,15 @@ export default class JquerySapFixHintsGenerator {
 
 			// Check whether the return value of the call expression is assigned to a variable,
 			// passed to another function or used elsewhere.
-			let isAssignmentStatement = false;
-			while (current && !isAssignmentStatement) {
+			let isExpectedValue = false;
+			while (current && !isExpectedValue) {
 				if (ts.isVariableDeclaration(current) ||
 					ts.isBinaryExpression(current) ||
 					ts.isVariableStatement(current) ||
 					ts.isConditionalExpression(current) ||
 					ts.isParenthesizedExpression(current) ||
+					ts.isReturnStatement(current) ||
+					ts.isArrowFunction(current) ||
```

Comment:
This now means that no log statement with an arrow function is migrated, which is more restrictive than needed.
I think this check is currently too simple by checking all parents until one match is found. In reality the check is a bit more complex as some (direct) parents indicate that the return value is not being used, and the lookup could stop then.

Extracting this functionality e.g. into our existing utils module and adding isolated unit tests for it would help implementing the logic without having to add too many actual cases within out jQuery.sap.log fixture.

Reply:
As long as there are no cases where we replace a call that should not be replaced, I'm fine with doing this change in a follow-up PR.